xml_converter/generators/proto_templates/node.proto

```diff
@@ -0,0 +1,274 @@
+        Category action_show_category = 13;
+        Category action_toggle_category = 14;
+
+        message guid{
+            int32 guid = 1;
+        }
+
+        enum reset_behavior {
+            always_visible = 0;
+            map_change = 1;
+            daily_reset = 2;
+            never = 3;
+            timer = 4;
+            map_reset = 5;
+            instance_change = 6;
+            daily_reset_per_character = 7;
+            weekly_reset = 8;
+        }
+    }
+
+}
+
+message Trail {
+    fixed32 achievement_bit = 1;
+    int32 achievement_id = 2;
+    float alpha = 3;
+    float animation_speed = 4;   
+    bool can_fade = 5; 
+    float distance_fade_end = 6;
+    float distance_fade_start = 7;
+    bool is_wall = 8;
+    string bhdraft__schedule = 9;
+    float bhdraft__schedule_duration = 10;
+    float scale = 11;
+    Category category = 12;
+    int32 map_id = 13;
+
+    message color {
+        string hex = 1;
+    }
+    
+    enum cull_chirality {
+        none = 0;
+        clockwise = 1;
+        counter_clockwise = 2;
+    }
+
+    message festival_filter {
+        bool dragonbash = 1;
+        bool festival_of_the_four_winds = 2;
+        bool halloween = 3;
+        bool lunar_new_year = 4;
+        bool super_adventure_festival = 5;
+        bool wintersday = 6;
+        bool none = 7;
+    }
+
+    message guid{
+        int32 guid = 1;
+    }
+
+    message map_type_filter {
+        bool unknown_map = 1;
+        bool redirect_map = 2;
+        bool character_create_map = 3;
+        bool pvp_map = 4;
+        bool gvg_map = 5;
+        bool instance_map = 6;
+        bool public_map = 7;
+        bool tournament_map = 8;
+        bool tutorial_map = 9;
+        bool user_tournament_map = 10;
+        bool center_map = 11;
+        bool eternal_battlegrounds_map = 12;
+        bool bluehome_map = 13;
+        bool blue_borderlands_map = 14;
+        bool green_home_map = 15;
+        bool green_borderlands_map = 16;
+        bool red_home_map = 17;
+        bool red_borderlands_map = 18;
+        bool fortunes_vale_map = 19;
+        bool jump_puzzle_map = 20;
+        bool obsidian_sanctum_map = 21;
+        bool edge_of_the_mists_map = 22;
+        bool public_mini_map = 23;
+        bool wvw_lounge_map = 24;
+    }
+    message mount_filter {
+        bool raptor = 1;
+        bool springer = 2;
+        bool skimmer = 3;
+        bool jackal = 4;
+        bool griffon = 5;
+        bool roller_beetle = 6;
+        bool warclaw = 7;
+        bool skyscalee = 8;
+        bool skiff = 9;
+        bool seige_turtle = 10;
+    }
+    message profession_filter {
+        bool guardian = 1;
+        bool warrior = 2;
+        bool engineer = 3;
+        bool ranger = 4;
+        bool thief = 5;
+        bool elementalist = 6;
+        bool mesmer = 7;
+        bool necromancer = 8;
+        bool revenantnt = 9;
+    }
+    message specialization_filter {
+    // Heart of Thorns Spec
```

Comment:
While clever to use the spec ids it does cause some level of confusion when presented in this proto file where most fields are ordered by their index. It's fine to continue using the spec ids here but they should be ordered by their id and any possibly future or past unused id that should not be used should be marked as `reserved`. Otherwise it is possible for arenanet to allocate and id and for a future revision of this code to have allocated an id, and then for those ids to collide causing this clever alignment to break down.

Reply:
Understood. Reworked indexes